### PR TITLE
Update to with-supertokens example

### PR DESCRIPTION
**Pull request by rishabhpoddar**
Adds cache control headers in the auth middleware to prevent issues with cached tokens being passed in GET API requests.

### CODE REVIEW

File: examples/with-supertokens/pages/auth/[[...path]].tsx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import Head from 'next/head'
-import React, { useEffect } from 'react'
+import React, { ComponentType, useEffect } from 'react'
```
Comment (rishabhpoddar) [inline examples/with-supertokens/pages/auth/[[...path]].tsx:2]:
```suggestion
import React, { useEffect } from 'react'
```

**Review by ijjk — COMMENTED** [review]
Hi, please run `pnpm lint-fix` to correct the lint errors!

**Comment by rishabhpoddar** [comment]
> Hi, please run pnpm lint-fix to correct the lint errors!

Done! Thanks.

**Review by leerob — APPROVED** [review]

**Comment by leerob** [comment]
In the future, making it such that maintainers can suggest edits to PRs would make these go through much faster. For example, we need to get off a new push here so we can validate CI / run it, but I don't have access.

**Comment by rishabhpoddar** [comment]
> In the future, making it such that maintainers can suggest edits to PRs would make these go through much faster. For example, we need to get off a new push here so we can validate CI / run it, but I don't have access.

There is not option to allow maintainers to edit PRs for PRs created from an org account: https://github.com/orgs/community/discussions/5634. From next time onwards, i'll use my personal GH account to create PRs. Thanks!

**Comment by leerob** [comment]
I still can't kick off a new build here, we'll need to get a new PR from your personal account. Thank you.

**Comment by rishabhpoddar** [comment]
@leerob, here is the new PR: https://github.com/vercel/next.js/pull/56035 (you can edit it).

**Pull request closed without merging**